### PR TITLE
[feature](write)add metrics for flink load data

### PR DESCRIPTION
**Pull request by vinlee19**
# Proposed changes
When you import data to Apache Doris in real time , you can observe the import of some metrics in Flink webui metrics. The metrics are a concatenation of table, and Flink metrics, e.g. table_1_totalFlushLoadBytes. Below is a description of how to use several metrics:

<!DOCTYPE html><figure class="md-table-fig table-figure" cid="n10" mdtype="table" style="box-sizing: border-box; --tw-border-spacing-x: 0; --tw-border-spacing-y: 0; --tw-translate-x: 0; --tw-translate-y: 0; --tw-rotate: 0; --tw-skew-x: 0; --tw-skew-y: 0; --tw-scale-x: 1; --tw-scale-y: 1; --tw-pan-x: ; --tw-pan-y: ; --tw-pinch-zoom: ; --tw-scroll-snap-strictness: proximity; --tw-ordinal: ; --tw-slashed-zero: ; --tw-numeric-figure: ; --tw-numeric-spacing: ; --tw-numeric-fraction: ; --tw-ring-inset: ; --tw-ring-offset-width: 0px; --tw-ring-offset-color: #fff; --tw-ring-color: rgb(59 130 246 / 0.5); --tw-ring-offset-shadow: 0 0 #0000; --tw-ring-shadow: 0 0 #0000; --tw-shadow: 0 0 #0000; --tw-shadow-colored: 0 0 #0000; --tw-blur: ; --tw-brightness: ; --tw-contrast: ; --tw-grayscale: ; --tw-hue-rotate: ; --tw-invert: ; --tw-saturate: ; --tw-sepia: ; --tw-drop-shadow: ; --tw-backdrop-blur: ; --tw-backdrop-brightness: ; --tw-backdrop-contrast: ; --tw-backdrop-grayscale: ; --tw-backdrop-hue-rotate: ; --tw-backdrop-invert: ; --tw-backdrop-opacity: ; --tw-backdrop-saturate: ; --tw-backdrop-sepia: ; margin: 1.2em 0px; overflow-x: auto; max-width: calc(100% + 16px); padding: 0px; cursor: default; caret-color: rgb(51, 51, 51); color: rgb(51, 51, 51); font-family: &quot;Open Sans&quot;, &quot;Clear Sans&quot;, &quot;Helvetica Neue&quot;, Helvetica, Arial, &quot;Segoe UI Emoji&quot;, sans-serif; font-size: 16px; font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration: none;">
Name | Metric Type | Description
-- | -- | --
totalFlushLoadBytes | Counter | Number of bytes imported.
flushTotalNumberRows | Counter | Number of rows imported for total processing
totalFlushLoadedRows | Counter | Number of rows successfully imported.
totalFlushTimeMs | Counter | Number of Import completion time. Unit milli
totalFlushSucceededNumber | Counter | Number of times that the data-batch been successfully imported.
totalFlushFailedNumber | Counter | Number of times that the data-batch been failed.
totalFlushFilteredRows | Counter | Number of rows that do not qualify for data quality flushed
totalFlushUnselectedRows | Counter | Number of rows filtered by where condition flushed
beginTxnTimeMs | Histogram | The time cost for RPC to Fe to begin a transaction, Unit milliseconds.
putDataTimeMs | Histogram | The time cost for RPC to Fe to get a stream load plan, Unit milliseconds.
readDataTimeMs | Histogram | Read data time, Unit milliseconds.
writeDataTimeMs | Histogram | Write data time, Unit milliseconds.
commitAndPublishTimeMs | Histogram | The time cost for RPC to Fe to commit and publish a transaction, Unit milliseconds.
loadTimeMs | Histogram | Import completion time

</figure>
The monitoring metrics for Flink write are shown below:
截屏2024-01-05 15 37 00
截屏2024-01-05 15 47 45

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: flink-doris-connector/src/main/java/org/apache/doris/flink/sink/writer/DorisWriteMetrics.java [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+
+    @VisibleForTesting
+    public void register(
+            SinkWriterMetricGroup sinkMetricGroup, int subTaskId, int histogramWindowSize) {
+        totalFlushNumberTotalRows =
+                sinkMetricGroup.counter(
+                        String.format(
+                                METRIC_NAME_FORMAT,
+                                tableIdentifier,
+                                subTaskId,
```
Comment (JNSimba) [inline flink-doris-connector/src/main/java/org/apache/doris/flink/sink/writer/DorisWriteMetrics.java:117]: Do we need to add subTaskId? It seems to be there by default?

File: flink-doris-connector/src/main/java/org/apache/doris/flink/sink/writer/DorisWriteMetrics.java [hunk 2]
```diff
@@ -0,0 +1,383 @@
...
+
+    private void flushSuccessLoad(RespContent responseContent) {
+        Optional.ofNullable(responseContent.getLoadBytes()).ifPresent(totalFlushLoadBytes::inc);
+        Optional.ofNullable(responseContent.getNumberLoadedRows())
+                .ifPresent(totalFlushLoadedRows::inc);
+        Optional.ofNullable(responseContent.getNumberTotalRows())
+                .ifPresent(totalFlushNumberTotalRows::inc);
+        Optional.ofNullable(responseContent.getNumberFilteredRows())
+                .ifPresent(totalFlushFilteredRows::inc);
+        Optional.ofNullable(responseContent.getLoadTimeMs()).ifPresent(totalFlushTimeMs::inc);
```
Comment (JNSimba) [inline flink-doris-connector/src/main/java/org/apache/doris/flink/sink/writer/DorisWriteMetrics.java:231]: Is getLoadTimeMs also a counter-type metric?

**Comment by JNSimba** [comment]
Thanks for the @vinlee19 . Can you post some final monitoring screenshots?

File: flink-doris-connector/src/main/java/org/apache/doris/flink/sink/writer/DorisWriteMetrics.java (hunk 2 above)
Comment (vinlee19) [reply flink-doris-connector/src/main/java/org/apache/doris/flink/sink/writer/DorisWriteMetrics.java:231]: There are two  metric type for LoadTimeMs:  counter and Historgram ,

**Comment by vinlee19** [comment]
> Thanks for the @vinlee19 . Can you post some final monitoring screenshots?

Yeah,I will post the final  monitoring screenshots later.

**Review by JNSimba — APPROVED** [review]
LGTM

**Pull request merged**